### DISCUSSION TITLE
If the credentials are in the code, can these be acceded by a stranger?

### DISCUSSION THREAD

**Discussion by hectorAguero** [incluud/accessible-astro-dashboard#10] [Q&A]
Hi, if I'm using this template in a private repo and deploy,

Can the credentials in the code be acceded by the browser devtools or something like that? I expect not.

Thanks for the template btw

**Comment by markteekman** [comment]
Hi @hectorAguero,

Thanks for the compliment and thanks for using the theme!

To answer your question: while code in your private repository is, well, private and can't be accessed by others, when you deploy the project, all client side JavaScript will be available for anyone through the developer console. This theme uses client side JavaScript and the `localStorage` purely as an example, but to make it secure, you should never put your own credentials in the code and use a secure server or database instead. I have some pointers in the `README.md` about this:

https://github.com/markteekman/accessible-astro-dashboard#login--authentication

I hope this helps! Let me know if you have any more questions 🙂 
Cheers,
Mark
